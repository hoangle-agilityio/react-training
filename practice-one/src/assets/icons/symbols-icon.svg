<svg xmlns="http://www.w3.org/2000/svg">
  <symbol width="42" height="42" id="twitter">
    <path d="M37.665 10.5066C36.4402 11.0349 35.1073 11.4152 33.7344 11.5633C35.1598 10.7165 36.2272 9.37865 36.7364 7.80083C35.3988 8.5965 33.9336 9.15444 32.4056 9.4499C31.7669 8.76716 30.9945 8.22326 30.1365 7.85209C29.2785 7.48092 28.3532 7.29043 27.4183 7.2925C23.6359 7.2925 20.5939 10.3585 20.5939 14.1209C20.5939 14.6493 20.6579 15.1776 20.762 15.6859C15.0983 15.3898 10.0471 12.684 6.68889 8.54131C6.07699 9.58645 5.75634 10.7765 5.76028 11.9875C5.76028 14.3571 6.96507 16.4464 8.80226 17.6752C7.71957 17.6326 6.66224 17.335 5.71626 16.8067V16.8907C5.71626 20.2089 8.06178 22.9587 11.1878 23.5911C10.6009 23.7435 9.99706 23.8215 9.39064 23.8232C8.94635 23.8232 8.52608 23.7792 8.1018 23.7192C8.96636 26.4249 11.484 28.3902 14.4819 28.4542C12.1364 30.2914 9.19851 31.3721 6.00845 31.3721C5.43607 31.3721 4.90773 31.3521 4.35938 31.2881C7.38534 33.2293 10.9757 34.3501 14.8422 34.3501C27.3943 34.3501 34.2628 23.9513 34.2628 14.9255C34.2628 14.6293 34.2628 14.3331 34.2428 14.0369C35.5716 13.0642 36.7364 11.8595 37.665 10.5066Z" fill="#EF2779"/>
  </symbol>
  <symbol id="facebook">
    <path d="M35.7321 4.81104H6.27302C5.56456 4.81104 4.99219 5.38341 4.99219 6.09187V35.551C4.99219 36.2594 5.56456 36.8318 6.27302 36.8318H35.7321C36.4406 36.8318 37.013 36.2594 37.013 35.551V6.09187C37.013 5.38341 36.4406 4.81104 35.7321 4.81104ZM32.0337 14.1571H29.4761C27.4708 14.1571 27.0825 15.1097 27.0825 16.5106V19.5966H31.8696L31.2452 24.4278H27.0825V36.8318H22.0913V24.4318H17.9166V19.5966H22.0913V16.0343C22.0913 11.8996 24.6169 9.64617 28.3073 9.64617C30.0765 9.64617 31.5934 9.77826 32.0377 9.8383V14.1571H32.0337Z" fill="#EF2779"/>
  </symbol>
  <symbol id="linkedin">
    <path d="M35.7165 4.81104H6.25739C5.54893 4.81104 4.97656 5.38341 4.97656 6.09187V35.551C4.97656 36.2594 5.54893 36.8318 6.25739 36.8318H35.7165C36.425 36.8318 36.9973 36.2594 36.9973 35.551V6.09187C36.9973 5.38341 36.425 4.81104 35.7165 4.81104ZM14.4747 32.0967H9.72364V16.8148H14.4747V32.0967ZM12.1012 14.7255C11.5565 14.7255 11.0241 14.564 10.5713 14.2614C10.1184 13.9588 9.76545 13.5287 9.55702 13.0255C9.34859 12.5223 9.29406 11.9686 9.40031 11.4344C9.50657 10.9003 9.76884 10.4096 10.154 10.0245C10.5391 9.63934 11.0298 9.37706 11.5639 9.27081C12.0981 9.16455 12.6518 9.21909 13.155 9.42751C13.6582 9.63594 14.0883 9.9889 14.3909 10.4418C14.6935 10.8946 14.855 11.427 14.855 11.9717C14.851 13.4927 13.6182 14.7255 12.1012 14.7255ZM32.2623 32.0967H27.5152V24.6639C27.5152 22.8908 27.4832 20.6133 25.0456 20.6133C22.576 20.6133 22.1957 22.5425 22.1957 24.5358V32.0967H17.4527V16.8148H22.0076V18.9042H22.0717C22.7041 17.7034 24.2531 16.4346 26.5666 16.4346C31.3777 16.4346 32.2623 19.6006 32.2623 23.7153V32.0967V32.0967Z" fill="#EF2779"/>
  </symbol>
  <symbol id="address">
    <g clip-path="url(#clip0_67_7968)">
    <path d="M13.0013 19.4816C16.4941 19.4816 19.3346 16.6411 19.3346 13.1483C19.3346 9.65544 16.4941 6.81494 13.0013 6.81494C9.50847 6.81494 6.66797 9.65544 6.66797 13.1483C6.66797 16.6411 9.50847 19.4816 13.0013 19.4816ZM13.0013 9.98161C14.7477 9.98161 16.168 11.4019 16.168 13.1483C16.168 14.8947 14.7477 16.3149 13.0013 16.3149C11.2549 16.3149 9.83464 14.8947 9.83464 13.1483C9.83464 11.4019 11.2549 9.98161 13.0013 9.98161Z" fill="#EF2779"/>
    <path d="M11.7485 31.8536C12.0164 32.045 12.3375 32.1479 12.6668 32.1479C12.9961 32.1479 13.3172 32.045 13.5851 31.8536C14.0665 31.5132 25.3794 23.3448 25.3335 13.1481C25.3335 6.16403 19.6509 0.481445 12.6668 0.481445C5.68272 0.481445 0.000139393 6.16403 0.000139393 13.1402C-0.0457773 23.3448 11.2671 31.5132 11.7485 31.8536ZM12.6668 3.64811C17.9061 3.64811 22.1668 7.90886 22.1668 13.156C22.2001 20.1829 15.2191 26.4924 12.6668 28.5619C10.1161 26.4909 3.13356 20.1797 3.16681 13.1481C3.16681 7.90886 7.42756 3.64811 12.6668 3.64811Z" fill="#EF2779"/>
    </g>
    <defs>
    <clipPath id="clip0_67_7968">
    <rect width="26" height="32" fill="white" transform="translate(0 0.314941)"/>
    </clipPath>
    </defs>
  </symbol>
  <symbol id="phone">
    <svg width="21" height="35" viewBox="0 0 21 35" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path d="M16.0417 1.27344H4.375C1.95417 1.27344 0 3.2276 0 5.64844V28.9818C0 31.4026 1.95417 33.3568 4.375 33.3568H16.0417C18.4625 33.3568 20.4167 31.4026 20.4167 28.9818V5.64844C20.4167 3.2276 18.4625 1.27344 16.0417 1.27344ZM13.125 30.4401H7.29167V28.9818H13.125V30.4401ZM17.8646 26.0651H2.55208V5.64844H17.8646V26.0651Z" fill="#EF2779"/>
    </svg>
  </symbol>
  <symbol id="email">
    <svg width="27" height="23" viewBox="0 0 27 23" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path d="M25.3125 1.00244H0.9375C0.418945 1.00244 0 1.42139 0 1.93994V20.6899C0 21.2085 0.418945 21.6274 0.9375 21.6274H25.3125C25.8311 21.6274 26.25 21.2085 26.25 20.6899V1.93994C26.25 1.42139 25.8311 1.00244 25.3125 1.00244ZM24.1406 4.24853V19.5181H2.10938V4.24853L1.30078 3.61865L2.45215 2.13916L3.70605 3.11475H22.5469L23.8008 2.13916L24.9521 3.61865L24.1406 4.24853ZM22.5469 3.11182L13.125 10.436L3.70312 3.11182L2.44922 2.13623L1.29785 3.61572L2.10645 4.24561L12.1143 12.0269C12.4021 12.2505 12.7561 12.3718 13.1206 12.3718C13.4851 12.3718 13.8391 12.2505 14.127 12.0269L24.1406 4.24853L24.9492 3.61865L23.7979 2.13916L22.5469 3.11182Z" fill="#EF2779"/>
    </svg>
  </symbol>
</svg>
