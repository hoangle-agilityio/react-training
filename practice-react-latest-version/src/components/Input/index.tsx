// Libs
import {
  Box,
  Flex,
  FormLabel,
  forwardRef,
  Input as ReactInput,
  BoxProps,
  FlexProps,
  InputProps,
  Textarea,
  TextareaProps,
  Text,
} from "@chakra-ui/react";
import * as React from "react";
import { memo, useId } from "react";

type StyleType = {
  wrapper?: BoxProps;
  inputWrap?: FlexProps;
  input?: InputProps & TextareaProps;
};

interface Props {
  label?: string;
  name: string;
  icon?: JSX.Element;
  iconPosition?: "left" | "right";
  styles?: StyleType;
  isTextArea?: boolean;
  error?: string;
}

const Input = forwardRef(
  (
    {
      label,
      name,
      icon,
      iconPosition,
      styles,
      isTextArea,
      error,
      ...rest
    }: Props,
    ref
  ) => {
    const id = useId();

    let iconStyle: FlexProps = {};
    // if exist icon, set style
    if (icon) {
      switch (iconPosition) {
        case "right":
          iconStyle = {
            paddingRight: "10px",
            flexDirection: "row-reverse",
          };
          break;
        default:
          iconStyle = {
            paddingLeft: "10px",
            flexDirection: "row",
          };
          break;
      }
    }

    return (
      <>
        <Box {...styles?.wrapper}>
          <FormLabel
            fontSize="base"
            fontFamily="heading"
            htmlFor={`${name}${id}`}
          >
            {label}
          </FormLabel>
          <Flex
            alignItems="center"
            border="1px"
            borderRadius="6"
            borderColor={error ? "red.200" : "gray.400"}
            {...iconStyle}
            {...styles?.inputWrap}
          >
            {icon}
            {isTextArea ? (
              <Textarea
                ref={ref}
                border="none"
                focusBorderColor="transparent"
                fontFamily="body"
                fontSize="base"
                id={`${name}${id}`}
                {...styles?.input}
                {...rest}
              />
            ) : (
              <ReactInput
                ref={ref}
                border="none"
                focusBorderColor="transparent"
                fontFamily="body"
                fontSize="base"
                id={`${name}${id}`}
                {...styles?.input}
                {...rest}
              />
            )}
          </Flex>
          {error && (
            <Text fontSize="base" color="red.200">
              {error}
            </Text>
          )}
        </Box>
      </>
    );
  }
);

export default memo(Input);
