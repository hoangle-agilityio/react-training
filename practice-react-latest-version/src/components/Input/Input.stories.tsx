// Libs
import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

// Components
import Input from "components/Input";
import { SearchIcon } from "@chakra-ui/icons";

export default {
  title: "Components/Input",
  component: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const InputBase = Template.bind({});
InputBase.args = {
  ...InputBase.args,
  label: "Name",
  styles: {
    wrapper: {
      width: "300px",
    },
  },
};

export const InputWithPlaceholder = Template.bind({});
InputWithPlaceholder.args = {
  placeholder: "Name",
  styles: {
    wrapper: {
      width: "300px",
    },
  },
};

export const InputWithIcon = Template.bind({});
InputWithIcon.args = {
  placeholder: "Search",
  icon: <SearchIcon />,
  styles: {
    wrapper: {
      width: "300px",
    },
  },
};

export const InputBaseWithError = Template.bind({});
InputBaseWithError.args = {
  ...InputBaseWithError.args,
  label: "Name",
  styles: {
    wrapper: {
      width: "300px",
    },
  },
  error: "This is an error.",
};

export const TextareaBase = Template.bind({});
TextareaBase.args = {
  ...TextareaBase.args,
  label: "Description",
  isTextArea: true,
  styles: {
    wrapper: {
      width: "300px",
    },
  },
};
