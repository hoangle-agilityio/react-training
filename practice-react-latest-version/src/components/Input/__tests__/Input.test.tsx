import { fireEvent, render } from "@testing-library/react";
import * as React from "react";
import Input from "..";
import { SearchIcon } from "@chakra-ui/icons";

const props = {
  label: "Name",
  name: "name",
  "aria-label": "Name",
};

describe("Input Component", () => {
  test("Should match snapshoot for Input", () => {
    const dom = render(<Input {...props} />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Textarea", () => {
    const dom = render(<Input {...props} isTextArea={true} />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Input with icon", () => {
    const dom = render(<Input {...props} icon={<SearchIcon />} />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Input with error", () => {
    const dom = render(<Input {...props} error="This is an error." />);

    expect(dom).toMatchSnapshot();
  });

  test("Should handle onChange event of Input", () => {
    const { getByLabelText } = render(<Input {...props} />);
    const input = getByLabelText("Name") as HTMLInputElement;
    const value = "test input component";

    fireEvent.change(input, { target: { value } });
    expect(input.value).toBe(value);
  });

  test("Should handle onChange event of Textarea", () => {
    const { getByLabelText } = render(<Input {...props} isTextArea={true} />);
    const input = getByLabelText("Name") as HTMLInputElement;
    const value = "test textarea component";

    fireEvent.change(input, { target: { value } });
    expect(input.value).toBe(value);
  });
});
