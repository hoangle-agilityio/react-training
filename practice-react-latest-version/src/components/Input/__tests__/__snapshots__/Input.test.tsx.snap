// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should match snapshoot for Input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-dh0v2a"
      >
        <label
          class="chakra-form__label css-2dkmm4"
          for="name:r0:"
        >
          Name
        </label>
        <div
          class="css-188q4y8"
        >
          <input
            aria-label="Name"
            class="chakra-input css-oce1e7"
            id="name:r0:"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dh0v2a"
    >
      <label
        class="chakra-form__label css-2dkmm4"
        for="name:r0:"
      >
        Name
      </label>
      <div
        class="css-188q4y8"
      >
        <input
          aria-label="Name"
          class="chakra-input css-oce1e7"
          id="name:r0:"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Component Should match snapshoot for Input with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-dh0v2a"
      >
        <label
          class="chakra-form__label css-2dkmm4"
          for="name:r3:"
        >
          Name
        </label>
        <div
          class="css-1ceu8sn"
        >
          <input
            aria-label="Name"
            class="chakra-input css-oce1e7"
            id="name:r3:"
          />
        </div>
        <p
          class="chakra-text css-0"
        >
          This is an error.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dh0v2a"
    >
      <label
        class="chakra-form__label css-2dkmm4"
        for="name:r3:"
      >
        Name
      </label>
      <div
        class="css-1ceu8sn"
      >
        <input
          aria-label="Name"
          class="chakra-input css-oce1e7"
          id="name:r3:"
        />
      </div>
      <p
        class="chakra-text css-0"
      >
        This is an error.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Component Should match snapshoot for Input with icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-dh0v2a"
      >
        <label
          class="chakra-form__label css-2dkmm4"
          for="name:r2:"
        >
          Name
        </label>
        <div
          class="css-19ep17b"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
              fill="currentColor"
            />
          </svg>
          <input
            aria-label="Name"
            class="chakra-input css-oce1e7"
            id="name:r2:"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dh0v2a"
    >
      <label
        class="chakra-form__label css-2dkmm4"
        for="name:r2:"
      >
        Name
      </label>
      <div
        class="css-19ep17b"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
            fill="currentColor"
          />
        </svg>
        <input
          aria-label="Name"
          class="chakra-input css-oce1e7"
          id="name:r2:"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input Component Should match snapshoot for Textarea 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-dh0v2a"
      >
        <label
          class="chakra-form__label css-2dkmm4"
          for="name:r1:"
        >
          Name
        </label>
        <div
          class="css-188q4y8"
        >
          <textarea
            aria-label="Name"
            class="chakra-textarea css-oce1e7"
            id="name:r1:"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dh0v2a"
    >
      <label
        class="chakra-form__label css-2dkmm4"
        for="name:r1:"
      >
        Name
      </label>
      <div
        class="css-188q4y8"
      >
        <textarea
          aria-label="Name"
          class="chakra-textarea css-oce1e7"
          id="name:r1:"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
